src/SAWScript/MGU.hs

```diff
@@ -293,21 +295,39 @@ newTypeIndex = do
   TI $ put $ rw { nameGen = nameGen rw + 1 }
   return $ nameGen rw
 
-newType :: TI Type
-newType = TyUnifyVar <$> newTypeIndex
-
+-- Construct a new type variable.
+--
+-- Collect the position that prompted us to make it; for example, if
+-- we're the element type of an empty list we get the position of the
+-- []. We haven't inferred anything, so use the Placeholder position.
```

Comment:
What is `Placeholder` in this context? Based on the capitalization, I expected it to be a data constructor for `Pos`, but there doesn't appear to be any data constructor with that name. (There are other references to "placeholder" elsewhere in this PR, and I'm not sure if they're referring to the same thing or not.)

Reply:
Edit sausage. It should be ```InfFresh```